### DISCUSSION TITLE
AddOrUpdate Method For Generics Dictionary

### DISCUSSION THREAD

**Discussion by suleman-khalid** [dotnet/runtime#74812] [General] [answered]
If we have a generics dictionary and we want to 
1- update if the key exists
2- add if the key doesn't exist

We currently have to use 
if ContainsKey
{ update }
else
{ add }

Can we add a method "AddOrUpdate" to do the above logic in one call?

**Comment by GalaxiaGuy** [comment]
```csharp
dictionary[key] = value;
```

https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.idictionary-2.item?view=net-6.0#remarks

> You can also use the `Item[]` property to add new elements by setting the value of a key that does not exist in the dictionary; for example, `myCollection["myNonexistentKey"] = myValue` in C#. However, if the specified key already exists in the dictionary, setting the `Item[]` property overwrites the old value.

**Comment by svick** [comment, marked as answer]
This is a question for dotnet/runtime, where it was discussed a few times (https://github.com/dotnet/runtime/issues/26848, https://github.com/dotnet/runtime/issues/1221) and ended up being closed, pointing to [`CollectionsMarshal.GetValueRefOrAddDefault()`](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.collectionsmarshal.getvaluereforadddefault) if you're looking for this method to improve performance.
